lib/stan_math_2.6.3/stan/math/prim/scal/prob/wiener_log.hpp

```diff
@@ -0,0 +1,224 @@
+      check_bounded(function, "A-priori bias", beta , 0, 1);
+      check_consistent_sizes(function,
+       "Random variable", y,
+       "Boundary separation", alpha,
+       "A-priori bias", beta,
+       "Nondecision time", tau,
+       "Drift rate", delta);
+
+      size_t N = std::max(max_size(y, alpha, beta), max_size(tau, delta));
+      if (!N)
+        return 0.0;
+      VectorView<const T_y> y_vec(y);
+      VectorView<const T_alpha> alpha_vec(alpha);
+      VectorView<const T_beta> beta_vec(beta);
+      VectorView<const T_tau> tau_vec(tau);
+      VectorView<const T_delta> delta_vec(delta);
+
+      if (!include_summand<propto, T_y, T_alpha, T_tau,
+          T_beta, T_delta>::value) {
+        return 0;
+      }
+
+      for (size_t i = 0; i < N; i++)
+        if (y_vec[i] < tau_vec[i]) {
+          lp = stan::math::negative_infinity();
+          return lp;
+        }
+
+        for (size_t i = 0; i < N; i++) {
+          typename scalar_type<T_beta>::type one_minus_beta = 1.0 - beta_vec[i];
+          typename scalar_type<T_alpha>::type alpha2 = alpha_vec[i] *
+                                                       alpha_vec[i];
+          T_return_type x = y_vec[i];
+          T_return_type kl, ks, tmp = 0;
+          T_return_type k, K;
+
+
+        x = x - tau_vec[i];  // remove non-decision time from x
+        x = x / alpha2;  // convert t to normalized time tt
+
+        // calculate number of terms needed for large t:
+        // if error threshold is set low enough
+        if (MY_PI * x * WIENER_ERR < 1) {
+          // compute bound
+          kl = sqrt(-2.0 * log(MY_PI * x * WIENER_ERR) / (pow(MY_PI, 2) * x));
+          // ensure boundary conditions met
+          kl = (kl > 1.0 / (MY_PI * sqrt(x))) ? kl : 1.0 / (MY_PI * sqrt(x));
+        } else {  // if error threshold set too high
+          kl = 1.0 / (MY_PI * sqrt(x));  // set to boundary condition
+        }
+        // calculate number of terms needed for small t:
+        // if error threshold is set low enough
+        if ((2.0 * sqrt(2.0 * MY_PI * x) * WIENER_ERR) < 1) {
+          // compute bound
+          ks = 2.0 + sqrt(-2.0 * x *
+               log(2.0 * sqrt(2.0 * MY_PI * x) * WIENER_ERR));
+          // ensure boundary conditions are met
+          ks = (ks > sqrt(x) + 1.0) ? ks : sqrt(x) + 1.0;
+        } else {  // if error threshold was set too high
+          ks = 2.0;  // minimal kappa for that case
+        }
+        // compute density: f(tt|0,1,w)
+        if (ks < kl) {  // if small t is better (i.e., lambda<0)
+          K = ceil(ks);  // round to smallest integer meeting error
+          for (k = -floor((K - 1.0) / 2.0); k <= ceil((K - 1.0) / 2.0); k++)
+            // increment sum
+            tmp += (one_minus_beta + 2.0 * k) *
+                    exp(-(pow((one_minus_beta + 2.0 * k), 2)) / 2.0 / x);
+            // add constant term
+            tmp = log(tmp) - 0.5 * LOG_TWO - MY_LN_SQRT_PI - 1.5 * log(x);
```

Comment:
At least bracket all of `0.5 * LOG_TWO - MY_LN_SQRT_PI` so it's evaluated as a double by itself.  You don't want to pollute potential non-constants in arithmetic.  Ideally, you'd just make it a constant.